account_payment_other_company/tests/test_account_payment.py

```diff
@@ -0,0 +1,111 @@
+        convert_file(
+            cls.cr, module,
+            get_resource_path(module, "tests",
+                              "test_account_payment_data.xml"),
+            None, 'init', False, 'test', cls.registry._assertion_report,
+        )
+        cls.account_obj = cls.env['account.account']
+        cls.invoice_obj = cls.env.ref(
+            'account_payment_other_company.customer_invoice_company_a')
+        cls.invoice_company_a = cls.env.ref(
+            'account_payment_other_company.customer_invoice_company_a')
+        cls.company_a = cls.env.ref(
+            'account_payment_other_company.company_a')
+        cls.company_b = cls.env.ref(
+            'account_payment_other_company.company_b')
+        cls.account_payment_obj = cls.env['account.payment']
+        cls.journal_1 = cls.env['account.journal']
+        cls.journal_2 = cls.env['account.journal']
+        cls.account_dt1 = cls.env['account.account']
+        cls.account_dt2 = cls.env['account.account']
+        cls.account_df1 = cls.env['account.account']
+        cls.account_d21 = cls.env['account.account']
+
+        cls.company_a.due_fromto_payment_journal_id = cls.env.ref(
+            'account_payment_other_company.sales_journal_company_a'
+        )
+        cls.company_b.due_fromto_payment_journal_id = cls.env.ref(
+            'account_payment_other_company.bank_journal_company_b'
+        )
+
+        cls.company_b.due_fromto_payment_journal_id.\
+            default_debit_account_id = cls.env.ref(
+                'account_payment_other_company.a_expense_company_b'
+            )
+
+        cls.company_a.due_from_account_id = cls.env.ref(
+            'account_payment_other_company.a_recv_company_a'
+        )
+        cls.company_a.due_to_account_id = cls.env.ref(
+            'account_payment_other_company.a_pay_company_a'
+        )
+        cls.company_b.due_from_account_id = cls.env.ref(
+            'account_payment_other_company.a_recv_company_b'
+        )
+        cls.company_b.due_to_account_id = cls.env.ref(
+            'account_payment_other_company.a_pay_company_b'
+        )
+
+        cls.chart = cls.env['account.chart.template'].search([], limit=1)
+        if not cls.chart:
+            raise ValidationError(
+                # translation to avoid pylint warnings
+                _("No Chart of Account Template has been defined !"))
+
+    def test_register_payment(self):
+        self.invoice_obj.action_invoice_open()
+        vals = {
+            'amount': self.invoice_obj.amount_total,
+            'journal_id': self.company_a.due_fromto_payment_journal_id.id,
+            'company_id': self.company_a.id,
+            'payment_type': 'outbound',
+            'currency_id': self.invoice_obj.currency_id.id,
+            'payment_date': self.invoice_obj.date_invoice,
+            'communication': 'findme',
+            'other_journal_id': self.company_b.
+            due_fromto_payment_journal_id.id,
```

Comment:
@osi-scampbell This can't happen because of https://github.com/OCA/multi-company/pull/177/files#diff-142ae26a307931583935b5f3cf785f90R25